ingestion/src/metadata/ingestion/source/dashboard/metabase/client.py

```diff
@@ -146,22 +147,81 @@ def get_collections_list(self) -> List[MetabaseCollection]:
             logger.warning("Failed to fetch the collections list")
         return []
 
+    def get_charts_dict(self) -> Dict:
+        charts_dict = {}
+        try:
+            resp_charts = self.client.get("/card")
+            if resp_charts:
+                for chart_data in resp_charts:
+                    chart = MetabaseChart.model_validate(chart_data)
+                    charts_dict[chart.id] = chart
+            return charts_dict
+        except Exception:
+            logger.debug(traceback.format_exc())
+            logger.warning("Failed to fetch the cards")
+        return {}
+
+    def _create_default_dashboard_details(
+        self, orphan_charts_id: List
+    ) -> MetabaseDashboardDetails:
+        """
+        Returns:
+            MetabaseDashboardDetails object representing the default dashboard containing orphaned charts
+        """
+        dashboard_data = {
+            "description": "Contains charts not associated with any dashboard",
+            "name": "Default Dashboard",
+            "id": "-1",
+            "card_ids": orphan_charts_id,
+        }
+        return MetabaseDashboardDetails(**dashboard_data)
+
+    def _process_dashboard_response(
+        self, resp_dashboard: Dict, charts_dict: Dict, dashboard_id: str
+    ) -> MetabaseDashboardDetails:
+        """
+        Process dashboard response and create MetabaseDashboardDetails object
+        """
+        if "ordered_cards" in resp_dashboard:
+            resp_dashboard["dashcards"] = resp_dashboard["ordered_cards"]
+
+        card_ids = []
+        for card in resp_dashboard["dashcards"]:
+            if card.get("card") and card["card"].get("id"):
+                card_id = str(card["card"]["id"])
+                card_ids.append(card_id)
+                if card_id in charts_dict:
+                    charts_dict[card_id].dashboard_ids.append(dashboard_id)
+
+        dashboard_data = {
+            "description": resp_dashboard.get("description"),
+            "name": resp_dashboard.get("name"),
+            "id": resp_dashboard.get("id"),
+            "creator_id": resp_dashboard.get("creator_id"),
+            "collection_id": resp_dashboard.get("collection_id"),
+            "card_ids": card_ids,
+        }
+
+        return MetabaseDashboardDetails(**dashboard_data)
```

Comment:
why are we creating a dict and then converting it to model we can directly create a model 

```
MetabaseDashboardDetails(
description = resp_dashboard.get("description"),
....
```